ui/plugins/eu.esdihumboldt.hale.ui/src/eu/esdihumboldt/hale/ui/io/schema/SourceSchemaActionUIAdvisor.java

```diff
@@ -0,0 +1,63 @@
+
+package eu.esdihumboldt.hale.ui.io.schema;
+
+import eu.esdihumboldt.hale.common.align.model.Type;
+import eu.esdihumboldt.hale.common.instance.model.DataSet;
+import eu.esdihumboldt.hale.common.schema.SchemaSpaceID;
+import eu.esdihumboldt.hale.common.schema.model.Schema;
+
+/**
+ * Source Schema action UI advisor.
+ * 
+ * @author Kapil Agnihotri.
+ */
+public class SourceSchemaActionUIAdvisor extends AbstractSchemaActionUIAdvisor<Schema> {
+
+	/**
+	 * @see eu.esdihumboldt.hale.ui.io.action.ActionUIAdvisor#supportsRemoval(java.lang.String)
+	 */
+	@Override
+	public boolean supportsRemoval(String resourceId) {
+		boolean supportRemoval = true;
+		Schema schema = schemaService.getSchema(resourceId, SchemaSpaceID.SOURCE);
+
+		// check if the source schema is associated with any instance, then
+		// it can't be removed.
+		supportRemoval = !instanceService.getInstanceTypes(DataSet.SOURCE).stream()
+				.anyMatch(type -> schema.getType(type.getName()) != null);
+
+		// if the data is loaded then return false. As from the alignment
+		// condition it might become true if the alignment is not
+		// loaded.
+		if (!supportRemoval)
+			return supportRemoval;
```

Comment:
I'd be in favour of adding braces also for one-line blocks. This is not done consistently throughout the codebase, though. @stempler What do you think?

Reply:
Even I am in the favour of adding braces, just observed in the code that at places they weren't so followed the same principle. 
Now I have added the braces.